From eb85b6bbf1c064a05fab0f0ec50912007a100d36 Mon Sep 17 00:00:00 2001
From: Xia Fukun <xiafukun@huawei.com>
Date: Tue, 15 Nov 2022 19:47:00 +0800
Subject: [PATCH 0034/1398] [Huawei] kernel/sched: Use
 DECLARE_PER_CPU_SHARED_ALIGNED declare rq_cputimes
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188002

--------------------------------

The following warnings are prompted when compiling the kernel：

kernel/sched/cputime.c:1134:38: warning: ignoring attribute ‘section
(".data..percpu..shared_aligned")’
because it conflicts with previous ‘section (".data..percpu")’
 [-Wattributes] 1134 | DEFINE_PER_CPU_SHARED_ALIGNED
(struct rq_cputime, rq_cputimes);
|                                      ^~~~~~~~~~

This is because the declaration and definition of the rq_cputimes
are inconsistent.
Use DECLARE_PER_CPU_SHARED_ALIGNED declare rq_cputimes.

Signed-off-by: Xia Fukun <xiafukun@huawei.com>
---
 kernel/sched/sched.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 252977131b94..0ee6b011fe35 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -1145,7 +1145,7 @@ struct rq_cputime {
 	struct cputime cpu_last_time;
 };
 
-DECLARE_PER_CPU(struct rq_cputime, rq_cputimes);
+DECLARE_PER_CPU_SHARED_ALIGNED(struct rq_cputime, rq_cputimes);
 
 static inline int cpu_of(struct rq *rq)
 {
-- 
2.17.1

